### PR TITLE
fix(#8293) add debug information when CHT Docker helper fails and exits

### PR DESCRIPTION
**Pull request by mrjones-plip**
# Description

This PR primary function is to add more info to CHT4 docker helper when it fails. this hopefully will enable further trouble shooting.

Specifically, this PR:
* adds new function: `service_has_image_downloaded()`,   `service_has_container()`, `container_status()`, `get_load_avg()` , `get_running_container_count()`, `get_global_running_container_count()` and `get_system_and_docker_info()`
* lets users pass in `DEBUG=true` to show the output from `get_system_and_docker_info()`
* will run  `get_system_and_docker_info()` and `clear` every second to give the users and up to date status of the script

closes medic/cht-core#8293

# Code review checklist

- [X] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [X] Tested: Unit and/or e2e where appropriate
- [X] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:cht4-compose-helper-debug/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:cht4-compose-helper-debug/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:cht4-compose-helper-debug/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
Here's a video of the script running in debug mode: 

https://github.com/medic/cht-core/assets/8253488/cb3de935-740c-4d8a-b078-a10459492315

**Comment by garethbowen** [comment]
@mrjones-plip Is this still worth doing or should we close it as out of date?

**Comment by mrjones-plip** [comment]
Well - not my finest work, but it is an improvement - thanks for the nudge @garethbowen .  I merged `master` to my branch, resolved conflicts and [added docs](https://github.com/medic/cht-docs/pull/1276).

File: scripts/docker-helper-4.x/cht-docker-compose.sh [hunk 1]
```diff
@@ -140,12 +154,91 @@ required_apps_installed(){
...
+
+get_system_and_docker_info(){
+  info='Service Status Container Image'
+  services="cht-upgrade-service haproxy healthcheck api sentinel nginx couchdb"
+  IFS=' ' read -ra servicesArray <<<"$services"
+  for service in "${servicesArray[@]}"; do
+    image=$(service_has_image_downloaded ${service})
+    container=$(service_has_container ${service})
+    status=$(container_status ${container})
+#    echo "${service}: image=${image} container=${container} status=${status}"
```
Comment (garethbowen) [inline scripts/docker-helper-4.x/cht-docker-compose.sh:232]: Worth dropping this? It looks like it's left over from an earlier version.

File: scripts/docker-helper-4.x/cht-docker-compose.sh [hunk 2]
```diff
@@ -140,12 +154,91 @@ required_apps_installed(){
...
+
+container_status(){
+	contianer=$1
+	status=$(docker inspect --format="{{.State.Status}}" "$contianer" 2>/dev/null)
+  if [ $status ];then
+    echo ${status}
+  else
+    echo "NA"
+  fi
+}
```
Comment (garethbowen) [inline scripts/docker-helper-4.x/cht-docker-compose.sh:202]: There's a little indentation weirdness here - mixing tabs and spaces maybe?

**Review by garethbowen — APPROVED** [review]
Definite improvement, thanks!

Two suggestions inline, and it looks like you still have a conflict to resolve.

**Comment by mrjones-plip** [comment]
Conflicts hopefully resolved in 227a7e064 !

File: scripts/docker-helper-4.x/cht-docker-compose.sh (hunk 2 above)
Comment (mrjones-plip) [reply scripts/docker-helper-4.x/cht-docker-compose.sh]: yikes! Thanks for the catch - fixed in b41a52d79

File: scripts/docker-helper-4.x/cht-docker-compose.sh (hunk 1 above)
Comment (mrjones-plip) [reply scripts/docker-helper-4.x/cht-docker-compose.sh]: dropped!

**Review by garethbowen — APPROVED** [review]
Fantastic, thanks!

**Pull request merged**